### ISSUE TITLE
username variable issues with ansible > 2.2.0.0

### ISSUE THREAD

**Issue by jduncan-rva** [ansible/lightbulb#216]
The following task in manage_ec2_instances/tasks/create.yml:

```
- name: Set local username to create on instances
  set_fact:
    username: "{{ item | regex_replace('.*-(\\w*)-\\w*$','\\1') }}"
  with_items: "{{ groups.lab_hosts }}"
  delegate_to: "{{ item }}"
  delegate_facts: yes
  tags:
    - always
    - provision
```

It looks like this regex fails on any newer version of ansible, this causes the user group and user tasks to fail later. 

Can we refactor this to work with newer versions, 2.2.2.0 which is required for the OpenShift installer.

**Comment by tima** [comment]
What version are you running and what is the error? Is it this one: https://github.com/ansible/lightbulb/issues/112?

**Comment by jduncan-rva** [comment]
Gerald set me straight. I was using 2.2.2.0 (OpenShift installer dependency). When I hit the error I downgraded because I knew it would work. Gerald informed me 2.4 is solid now.

sorry for the confusion.

**Comment by tima** [comment]
@jduncan-rva: Glad you got that sorted out. Yeah we got bit by a core engine bug where setting a fact using delegation broke when they were fixing a CVE in 2.2.1. It was finally fixed in 2.3.2.
